python/plugins/processing/algs/gdal/gdaladdo.py

```diff
@@ -125,8 +132,9 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
         arguments = []
         arguments.append(fileName)
 
-        arguments.append('-r')
-        arguments.append(self.methods[self.parameterAsEnum(parameters, self.RESAMPLING, context)][1])
+        if self.RESAMPLING in parameters and parameters[self.RESAMPLING] is not None:
+            arguments.append('-r')
```

Comment:
Does this also need a version check?

Reply:
Not AFAIK, document doesn't link this parameter's default value to a specific gdal version. In fact, even the algorithm itself meant for it to be optional, it was just not reflected in the resulting command line.

Reply:
@nyalldawson , here, gdal 2.2's gdaladdo does indeed have a default value and doesn't need the presence of -r:

https://github.com/OSGeo/gdal/blob/release/2.2/gdal/apps/gdaladdo.cpp#L111